### PR TITLE
Simplify _get_xdg_cache_dir in setupext.

### PR DESCRIPTION
**Pull request by anntzer**
1. we don't support google app engine anymore (#8939).
2. in setupext, the cache dir is only used to cache the freetype
   tarball, so it's really not a huge deal if a cache dir cannot be
   located (this case is handled explicitly by the freetype build code).

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: setupext.py [hunk 1]
```diff
@@ -23,43 +23,18 @@
...
     """
-    path = os.environ.get('XDG_CACHE_HOME')
-    if path is None:
-        path = _get_home()
-        if path is not None:
-            path = os.path.join(path, '.cache', 'matplotlib')
-    return path
+    cache_dir = os.environ.get('XDG_CACHE_HOME')
+    if not cache_dir:
+        cache_dir = os.path.expanduser('~/.cache/matplotlib')
```
Comment (jklymak) [inline setupext.py]: Does this work Win32?  Otherwise looks good to me.

Comment (anntzer) [reply setupext.py]: https://docs.python.org/3/library/os.path.html#os.path.expanduser yes

File: setupext.py [hunk 2]
```diff
@@ -23,43 +23,18 @@
...
+
+    See http://standards.freedesktop.org/basedir-spec/basedir-spec-latest.html
     """
-    path = os.environ.get('XDG_CACHE_HOME')
-    if path is None:
-        path = _get_home()
-        if path is not None:
-            path = os.path.join(path, '.cache', 'matplotlib')
-    return path
+    cache_dir = os.environ.get('XDG_CACHE_HOME')
```
Comment (QuLogic) [inline setupext.py]: Shouldn't this add `matplotlib` on the end?

File: setupext.py [hunk 3]
```diff
@@ -23,43 +23,18 @@
...
-
-
 def _get_xdg_cache_dir():
     """
-    Returns the XDG cache directory, according to the `XDG
-    base directory spec
-    <http://standards.freedesktop.org/basedir-spec/basedir-spec-latest.html>`_.
+    Return the XDG cache directory.
+
+    See http://standards.freedesktop.org/basedir-spec/basedir-spec-latest.html
```
Comment (QuLogic) [inline setupext.py]: https

File: setupext.py (hunk 2 above)
Comment (anntzer) [reply setupext.py]: Well, the previous version did not, but that was arguably a bug... fixed.

**Pull request merged**